### DISCUSSION TITLE
Dependency on Configurations of Targets

### DISCUSSION THREAD

**Discussion by veimox** [nrwl/nx#9236] [Feature Requests]
I have 3 packages. 2 of them define the `production` configuration in the `build` target. The last one is dependent on the other two packages does not define `production` but has instead two other configs like `slow` and `fast`. When I run the `build:slow`  config it builds the default configuration of the dependees, but I would like to run the `production` configuration instead. [I have found that you can explicitly or implicitly](https://nx.dev/configuration/projectjson#targets) specify dependencies on *targets* but can you do that for configurations too? If not it would be a great addition.

* Package A
  - `default`
  - `production`
* Package B
  - `default`
  - `production`
* Package C
  - `slow`
  - `fast`

**Comment by richard-ling** [comment]
I have a similar problem. Or maybe exactly the same problem!

We have 3 configs for our apps: `prod`, `dev`, `qa`.  We only have 2 configs for libs,  since `dev` and `qa` are the same.

When I build our apps using `--configuration=qa`, I'd like the libs to build with `--configuration=dev`.  There seems to be no way to do it using the configuration files.

So at the moment we have a `qa` config for the libs, which builds exactly the same binaries as the `dev` config, which adds needless time and cost to our CI pipeline.  We also have 2 separate sets of Nx caches which build the same thing.

**Comment by dmitry-stepanenko** [comment]
Related issue https://github.com/nrwl/nx/issues/15064
